test/metabase/driver/presto_test.clj

```diff
@@ -0,0 +1,134 @@
+(ns metabase.driver.presto-test
+  (:require [expectations :refer :all]
+            [toucan.db :as db]
+            [metabase.driver :as driver]
+            [metabase.driver.generic-sql :as sql]
+            [metabase.models.table :as table]
+            [metabase.test.data :as data]
+            [metabase.test.data.datasets :as datasets]
+            [metabase.test.util :refer [resolve-private-vars]])
+  (:import (metabase.driver.presto PrestoDriver)))
+
+(resolve-private-vars metabase.driver.presto details->uri details->headers quote-name quote+combine-names unprepare apply-page)
+
+;;; HELPERS
+
+(expect
+  "http://localhost:8080/"
+  (details->uri {:host "localhost", :port 8080, :ssl false} "/"))
+
+(expect
+  "https://localhost:8443/"
+  (details->uri {:host "localhost", :port 8443, :ssl true} "/"))
+
+(expect
+  "http://localhost:8080/v1/statement"
+  (details->uri {:host "localhost", :port 8080, :ssl false} "/v1/statement"))
+
+(expect
+  {"X-Presto-Source" "metabase"
+   "X-Presto-User"   "user"}
+  (details->headers {:user "user"}))
+
+(expect
+  {"X-Presto-Source"    "metabase"
+   "X-Presto-User"      "user"
+   "X-Presto-Catalog"   "test_data"
+   "X-Presto-Time-Zone" "America/Toronto"}
+  (details->headers {:user "user", :catalog "test_data", :report-timezone "America/Toronto"}))
+
+(expect
+  "\"weird.table\"\" name\""
+  (quote-name "weird.table\" name"))
+
+(expect
+  "\"weird . \"\"schema\".\"weird.table\"\" name\""
+  (quote+combine-names "weird . \"schema" "weird.table\" name"))
+
+(expect
+  ;; adapted from unprepare-test
+  "SELECT 'Cam''s Cool Toucan' FROM TRUE WHERE x ?? y AND z = from_iso8601_timestamp('2017-01-01T00:00:00.000Z')"
+  (unprepare "SELECT ? FROM ? WHERE x ?? y AND z = ?"
+             ["Cam's Cool Toucan", true, #inst "2017-01-01T00:00:00.000Z"]))
```

Comment:
no need to copy and paste the test, if you switch to using `metabase.driver.generic-sql.util.unprepare/unprepare` you don't need this test at all since that's tested elsewhere